libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt

```diff
@@ -225,6 +225,17 @@ class MapboxNavigationTest {
         verify(exactly = 1) { directionsSession.requestRoutes(any(), any()) }
     }
 
+    @Test
+    fun enhanced_location_used_for_reroute() {
+        val offRouteObserverSlot = slot<OffRouteObserver>()
+        verify { tripSession.registerOffRouteObserver(capture(offRouteObserverSlot)) }
+
+        offRouteObserverSlot.captured.onOffRouteStateChanged(true)
+
+        verify(exactly = 1) { tripSession.getEnhancedLocation() }
+        verify(exactly = 0) { tripSession.getRawLocation() }
```

Comment:
An ideal test would be to mock `adjustedRouteOptionsProvider` and verify that the exact location returned by `tripSession.getEnhancedLocation` is pushed to `adjustedRouteOptionsProvider.getRouteOptions(location)`. Not going to block, but just something to consider.

Reply:
Test added. Could you take a quick 👀 when you have a chance @LukasPaczos